tests/imagecheck.py

```diff
@@ -0,0 +1,193 @@
+        else:
+            print("ERROR", exc.returncode, exc.output, exc.stderr)
+            raise
+
+
+def get_test_list(filename):
+    """
+    Read YAML test configuration from a file
+    :param filename: str: path to YAML file with test settings for a single docker file
+    :return: [(command, expected_text)]: array of command and expected output pairs
+    """
+    with open(filename) as infile:
+        testinfo_list = []
+        unittest_config = yaml.load(infile)
+        for testinfo in unittest_config['commands']:
+            cmd = testinfo['cmd']
+            expect_text = testinfo['expect_text']
+            testinfo_list.append((cmd, expect_text))
+        return testinfo_list
+
+
+def build_docker_image(imagename, build_path):
+    print("Building docker image for path {}".format(build_path))
+    run_bash_cmd("docker build -t {} {}".format(imagename, build_path))
+
+
+def run_docker_get_output(imagename, cmd, workdir=None, user=None):
+    """
+    Launch docker process with run command passing the cmd.
+    :param imagename: str: name of the image to run
+    :param cmd: str: commmand to run inside the image
+    :param workdir: str: optional flag to run in a particular directory
+    :param user: str: optional flag to run as a particular user
+    :return: str: output of the docker process
+    """
+    options = ""
+    if workdir:
+        options += "--workdir {} ".format(workdir)
+    if user:
+        options += "--user {} ".format(user)
+    options += "-it "
+    print("Testing image {} with: docker run {} {}".format(imagename, options, cmd))
+    docker_cmd = "docker run {}{} {}".format(options, imagename, cmd)
+    return run_bash_cmd(docker_cmd, ignore_non_zero_exit_status=True)
+
+
+def run_tests(imagename, unittest_filepath):
+    """
+    Run all tests contained in a unittest_filename against imagename
+    :param imagename: str: name of the image to test
+    :param unittest_filepath: str path to unittest.yml file
+    :return: bool: true if we had an error
+    """
+    had_error = False
+    for cmd, expect_text in get_test_list(unittest_filepath):
+        expect_pattern = re.compile(expect_text, re.DOTALL)
+        docker_output = run_docker_get_output(imagename, cmd)
+        if not re.match(expect_pattern, docker_output):
+            print_test_error(cmd, expect_text, docker_output)
+            had_error = True
+        docker_output_with_options = run_docker_get_output(imagename, cmd, workdir=TEST_WORKDIR, user=TEST_USER)
+        if not re.match(expect_pattern, docker_output_with_options):
+            print_test_error(cmd + " (with workdir and user options)", expect_text, docker_output_with_options)
+            had_error = True
+    return had_error
+
+
+def print_test_error(cmd, expect_text, cmd_output):
+    print("Error running {}".format(cmd))
+    print("Expected Regex: {}".format(expect_text))
+    print("Actual: {}".format(cmd_output))
+
+
+def find_unittest_info():
```

Comment:
I don't see this called from anywhere except the test. Is it leftover from singularity/local registry testing?